docs: initial clion setup (#3548)
yugabyte/yugabyte-db: docs/content/latest/contribute/core-database/clion-setup.md

---
title: CLion project setup
linkTitle: CLion project setup
description: CLion project setup
image: /images/section_icons/index/quick_start.png
headcontent: CLion project setup.
type: page
menu:
  latest:
    identifier: clion-setup
    parent: contribute-db-core
    weight: 2912
isTocNested: true
showAsideToc: true
---

> There are two options for build system to use with YugabyteDB, cmake & ninja.

Reviewer: ```suggestion
There are two options for build systems that you can use with YugabyteDB, `cmake` and `ninja`.
```

* `make` is well-supported by CLion, but slower for rebuild comparing to ninja.

> * `ninja` is faster, but CLion has limited support for ninja (for example it doesnt allow to [rebuild individual files](https://youtrack.jetbrains.com/issue/CPP-17622))

Reviewer: ```suggestion
* `ninja` is faster, but CLion has limited support for `ninja` (for example, it doesn't allow you to [rebuild individual files](https://youtrack.jetbrains.com/issue/CPP-17622)).
```

> #### Following steps are needed for configuring CLion project for YugabyteDB:

Reviewer: ```suggestion
#### Configure a CLion project for YugabyteDB:
```

> 1. Use “File / Open…” to open the project root directory.

Reviewer: ```suggestion
1. Click **File > Open…** to open the project root directory.
```

> 2. Put build/debug-clang-dynamic (or build/debug-clang-dynamic-ninja if you want to use ninja) as a “Generation path” in “Preferences/Build, Execution, Deployment/CMake”: 

Reviewer: ```suggestion
2. Select `build/debug-clang-dynamic` (or `build/debug-clang-dynamic-ninja` if you want to use `ninja`) as the **Generation path** in **Preferences** > Build, Execution, Deployment/CMake”: 
```
Make GUI elements and titles bold.

> If you want to build with ninja, use build/debug-clang-dynamic-ninja as a generation path and add -G Ninja into “CMake options”:    

Reviewer: I'm having a hard time telling what is selected versus what is entered. And, bold the GUI elements. Put text to be entered in backticks.

Author: See changes
